### ISSUE TITLE
LogVerifier::VerifyResultString has FATAL log

### ISSUE THREAD

**Issue by Martin2112** [google/certificate-transparency#967]
Others might disagree with me but I think it's not ideal to have a simple method that maps an enum value to a string potentially crash the app.

It also makes it hard to avoid an ugly compiler warning on GCC, which isn't aware that LOG(FATAL) terminates.

I think this behaviour should be removed and required validation done by callers. In some cases there are currently nested LOG(FATAL) calls. e.g see server.h at line aound 320 so it would have a benefit there.

**Comment by pphaneuf** [comment]
I was going to reply something that disagrees with you, yes, but I've had another idea: I think this is an `enum` declared by us (as opposed to "generated by `protoc`")? If that's the case, I'd suggest turning it into an `enum class`, like I had temporarily done with `Cert::Status`, to make it a compile error to have a value that's not in the `enum`? :smiley: 

It might make it more acceptable to remove the `LOG(FATAL)`...

**Comment by Martin2112** [comment]
Yes it's our enum and it probably should be a c++11 enum. I think there are still ways around the compile time checks for the determined.

Do nested LOG(FATAL) even work properly? I suspect that message never gets fully logged.

**Comment by pphaneuf** [comment]
Nested?!? You mean, if a `LOG(FATAL)` is involved when generating another `LOG(FATAL)` because of an `enum` value? You'd only see the inner one, I suspect (the one about the `enum`)...

**Comment by Martin2112** [comment]
Yes it does LOG(FATAL) of the result of calling VerifyResultString on a bad value, which also has a LOG(FATAL).
